lim_test.py

```diff
@@ -0,0 +1,259 @@
+'''tests for the longintmath.py module'''
+import unittest
+import longintmath as lim
+
+class TestLIMFunc(unittest.TestCase):
+    '''tests for the long_int_(mathfunction) functions'''
+    def setUp(self):
+        self.long_big = 10**1000
+        self.float_big = 1.0e+300
+        self.float_sm = 1.0e-100
+
+    def tearDown(self):
+        del self.long_big
+        del self.float_big
+        del self.float_sm
+
+    def error_tester(self, result, expected):
+        '''use the inverse error to avoid trouble with overflow errors.'''
+        allowed_error = 1e-8
+        if result - expected == 0:
+            inv_error = float('inf')
+        else:
+            inv_error = abs(expected // (result - expected))
+        self.assertGreater(inv_error, 1/allowed_error)
+
+    def test_li_div(self):
+        '''test long_int_div'''
+        self.assertEqual(lim.long_int_div(self.float_big, self.long_big), 0)
+        self.assertEqual(lim.long_int_div(self.float_sm, self.float_big), 0)
+
+        result1 = lim.long_int_div(self.float_big, self.float_sm)
+        self.error_tester(result1, 10**400)
+
+        result2 = lim.long_int_div(self.long_big, -self.float_big)
+        self.error_tester(result2, -10**700)
+
+        result3 = lim.long_int_div(2, self.float_big)
+        self.error_tester(result3, 2e-300)
+
+        result4 = lim.long_int_div(-self.float_big, 5)
+        self.error_tester(result4, -2e+299)
+
+    def test_li_times(self):
+        '''test long_int_times'''
+        result1 = lim.long_int_times(self.long_big, self.float_big)
+        self.error_tester(result1, 10**1300)
+
+        result2 = lim.long_int_times(-4 * self.float_sm, self.float_sm)
+        self.error_tester(result2, -4e-200)
+    def test_li_pow(self):
+        '''test long_int_pow'''
+        result1 = lim.long_int_pow(self.long_big, 2.5)
+        self.error_tester(result1, 10**2500)
+
+
+class TestLITable(unittest.TestCase):
+    '''tests for all functions on a LongIntTable that aren't add and remove'''
+    def setUp(self):
+        self.table_a = lim.LongIntTable({3:3, 1:1, 2:2})
+        self.table_a1 = lim.LongIntTable({3:3, 1:1, 2:2})
+
+        self.table_b = lim.LongIntTable({1:1, 2:0, 3:3})
+        self.table_b1 = lim.LongIntTable({1:1, 2:0, 3:3})
+
+        self.table_c = lim.LongIntTable({1:1, 2:5, 3:3})
+
+        self.empty_table_a = lim.LongIntTable({})
+        self.empty_table_b = lim.LongIntTable({1:0, 2:0})
+
+    def tearDown(self):
+        del self.table_a
+        del self.table_a1
+
+        del self.table_b
+        del self.table_b1
+
+        del self.table_c
+
+        del self.empty_table_a
+        del self.empty_table_b
+
+    def test_values(self):
```

Comment:
test only what you need to, to validate things are functioning correctly and name things in a way the help you understand if you're missing an edge case.